services/s3/src/it/java/software/amazon/awssdk/services/s3/checksum/AsyncHttpChecksumIntegrationTest.java

```diff
@@ -234,6 +234,11 @@ void asyncValidSignedTrailerChecksumCalculatedBySdkClient() {
         assertThat(response).isEqualTo("Hello world");
     }
 
+    /**
+     * If http is used, payload signing will be enforced, but it's not currently supported in async path
+     * TODO: re-enable it once it's supported
+     */
+    @Disabled("Payload signing is not supported for S3 async client")
```

Comment:
This test is not supported today. The reason it passes is because payload signing is bypassed (request.contentStreamProvider() == null)

https://github.com/aws/aws-sdk-java-v2/blob/38e221bd815af31a6c6b91557499af155103c21a/core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AbstractAwsS3V4Signer.java#L279-L285